flatsurf/geometry/categories/euclidean_polygons.py

```diff
@@ -852,6 +852,64 @@ def get_point_position(self, point, translation=None):
 
             return PolygonPosition(PolygonPosition.OUTSIDE)
 
+        def join(self, other, edge, other_edge):
+            r"""
+            Return the polygon obtained by gluing this polygon and ``other``
+            along their ``edge`` and ``other_edge``, respectively.
+
+            The polygons have to be such that the glued edges are identical but
+            with opposite orientation.
+
+            INPUT:
+
+            - ``other`` -- a polygon over the same base ring as this polygon
+
+            - ``edge`` -- an integer; the index of the edge of this polygon
+              along which to glue
+
+            - ``other_edge`` -- an integer; the index of the edge of ``other``
+              along which to glue
+
+            EXAMPLES::
```

Comment:
It would be nice to have an example where `P` and `Q` indeed have an edge in common but do not define a valid polygon in the plane (ie has a self intersection).